pkg/test/echo/check/checkers.go

```diff
@@ -94,19 +95,52 @@ func Error() Checker {
 	}
 }
 
+// ErrorContains is similar to Error, but checks that the error message contains the given string.
+func ErrorContains(expected string) Checker {
+	return func(_ echo.Responses, err error) error {
+		if err == nil {
+			return errors.New("expected error, but none occurred")
+		}
+		if !strings.Contains(err.Error(), expected) {
+			return fmt.Errorf("expected error to contain %s: %v", expected, err)
+		}
+		return nil
+	}
+}
+
 func OK() Checker {
-	return Code(echo.StatusCodeOK)
+	return StatusCode(http.StatusOK)
 }
 
-func Code(expected string) Checker {
+// StatusCode checks that the response status code matches the expected value. If the expected value is zero,
+// checks that the response code is unset.
+func StatusCode(expected int) Checker {
+	expectedStr := ""
+	if expected > 0 {
+		expectedStr = strconv.Itoa(expected)
+	}
 	return Each(func(r echo.Response) error {
-		if r.Code != expected {
-			return fmt.Errorf("expected response code %s, got %q", expected, r.Code)
+		if r.Code != expectedStr {
+			return fmt.Errorf("expected response code `%s`, got %q", expectedStr, r.Code)
 		}
 		return nil
 	})
 }
 
+// TooManyRequests checks that at least one message receives a StatusTooManyRequests status code.
+func TooManyRequests() Checker {
```

Comment:
Not concerned yet but in general we should be conscious of localized Checkers vs ones here. If we end up with too many niche ones it will make usage harder.

Reply:
Agree in general. The ones here seem pretty general-purpose. TBH I'm not sure how many specialized ones we'll have at the end of the day. I guess we'll see as we continue to refactor.